File: LeetCode/1.py
# slow solution
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

# faster solution
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        retMap = {} 
        
        for i, n in enumerate(nums):
            diff = target - n
            if diff in retMap:
                return [retMap[diff], i]
            retMap[n] = i
        return
        

File: Problem_Solving/LeetCode/1.cpp

// Faster
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret(2);
        unordered_map<int, int> numbers;

        for(int i=0; i<nums.size(); ++i){
            int diff = target - nums[i];
            
            if(numbers.count(diff)){
                return {numbers[diff], i};
            }
            numbers[nums[i]] = i;
        }
        
        return {};
    }
};


// Slower
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        for(int i=0; i<nums.size(); ++i){
            for(int j=i+1; j<nums.size(); ++j){
                if(nums[i] + nums[j] == target){
                    ret.push_back(i);
                    ret.push_back(j);
                }
            }
        }
        return ret;
    }
};